src/code/z_map_exp.c

```diff
@@ -1,31 +1,255 @@
+        return false;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_exp/func_8010A2AC.s")
+    return true;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_exp/func_8010A2DC.s")
+s32 sDungeonScenes[] = {
+    SCENE_MITURIN, SCENE_HAKUGIN, SCENE_SEA, SCENE_INISIE_N, SCENE_INISIE_R,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_exp/func_8010A33C.s")
+/**
+ * If the current scene is a dungeon scene, this function returns an index
+ * indicating which one it is. Otherwise, it returns -1.
+ */
+s32 Map_GetDungeonAreaIndex(PlayState* play) {
+    s32 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_exp/func_8010A410.s")
+    for (i = 0; i < ARRAY_COUNT(sDungeonScenes); i++) {
+        if (Play_GetOriginalSceneNumber(play->sceneNum) == sDungeonScenes[i]) {
+            return i;
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_exp/func_8010A430.s")
+    return -1;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_exp/func_8010A54C.s")
+/**
+ * Returns true if the current scene is a dungeon scene, false otherwise.
+ */
+s32 Map_IsInDungeonArea(PlayState* play) {
+    if (Map_GetDungeonAreaIndex(play) == -1) {
+        return false;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_exp/func_8010A580.s")
+    return true;
+}
+
+s32 sBossScenes[] = {
+    SCENE_MITURIN_BS, // DUNGEON_INDEX_WOODFALL_TEMPLE
+    SCENE_HAKUGIN_BS, // DUNGEON_INDEX_SNOWHEAD_TEMPLE
+    SCENE_SEA_BS,     // DUNGEON_INDEX_GREAT_BAY_TEMPLE
+    SCENE_INISIE_BS,  // DUNGEON_INDEX_STONE_TOWER_TEMPLE
+};
+
+/**
+ * If the current scene is a boss scene, this function returns an index
+ * indicating which one it is. Otherwise, it returns -1.
+ */
+s32 Map_GetBossAreaIndex(PlayState* play) {
+    s32 i;
+
+    for (i = 0; i < ARRAY_COUNT(sBossScenes); i++) {
+        if (Play_GetOriginalSceneNumber(play->sceneNum) == sBossScenes[i]) {
+            return i;
+        }
+    }
+
+    return -1;
+}
+
+/**
+ * Returns true if the current scene is a boss scene, false otherwise.
+ */
+s32 Map_IsInBossArea(PlayState* play) {
+    if (Map_GetBossAreaIndex(play) == -1) {
+        return false;
+    }
+
+    return true;
+}
+
+s32 D_801BF5A4[] = {
+    SCENE_22DEKUCITY,
+    SCENE_KOEPONARACE,
+    SCENE_F01,
+};
+
+s32 func_8010A238(PlayState* play) {
+    s32 i;
+
+    for (i = 0; i < ARRAY_COUNT(D_801BF5A4); i++) {
+        if (Play_GetOriginalSceneNumber(play->sceneNum) == D_801BF5A4[i]) {
+            return i;
+        }
+    }
+
+    return -1;
+}
+
+s32 func_8010A2AC(PlayState* play) {
+    if (func_8010A238(play) == -1) {
+        return false;
+    }
+
+    return true;
+}
+
+void Map_SavePlayerInitialInfo(PlayState* play) {
+    Player* player = GET_PLAYER(play);
+
+    sPlayerInitialPosX = player->actor.world.pos.x;
+    sPlayerInitialPosZ = player->actor.world.pos.z;
+    sPlayerInitialDirection = (0x7FFF - player->actor.shape.rot.y) / 0x400;
+}
```

Comment:
I know this is what this is called in OoT, but I'm not sure I like this name very much. "PlayerInitial" very much makes me think of when "player" is initialized, which is when the scene is initialized. However, this function is called every single room transition, which happens many times after player is initialized. I think it could be something like:

`Map_SavePlayerInitialInfo` -> `Map_SavePlayerRoomInitInfo`
`sPlayerInitialPosX` -> `sPlayerRoomInitPosX`
`sPlayerInitialPosZ` -> `sPlayerRoomInitPosZ`
`sPlayerInitialDirection` -> `sPlayerRoomInitDirection` (or maybe `sPlayerRoomInitDir`)

I even wonder if this could reflect that these are the coordinates and direction of the red arrow/marker on the minimap that's present for each room:
`sMinimapInitPosX` or `sMinimapPlayerInitPosX` or `sMiniMapInitEntrancePosX` just to brainstorm a few names. But maybe just `RoomInit` is fine
![image](https://user-images.githubusercontent.com/47598039/177019906-5e6957f3-bfab-41e0-bf9b-5bdcc996b3ab.png)

Reply:
I just tested with some random values, and indeed these are for the red arrow.